From 96f42dc57eb8deffe226b88dafdc2dbe9a4a77d2 Mon Sep 17 00:00:00 2001
From: "Yin, Zhiye" <zhiyex.yin@intel.com>
Date: Tue, 15 Jan 2019 09:55:42 +0800
Subject: [PATCH 1149/1418] media: intel-ipu4: [ICI] start stream with correct
 parameters

Low brightness of first few frames if start stream without css
paramerters, lead to Camera CTS failure
android.hardware.camera2.cts.AllocationTest#testAllocationFromCameraFlexibleYuv

Change-Id: Ie9b608d0947c44c7b8a09476360c09628d1c9bac
Signed-off-by: Yin, Zhiye <zhiyex.yin@intel.com>
---
 drivers/media/pci/intel/ici/ici-isys-frame-buf.c | 6 ++++++
 drivers/media/pci/intel/ici/ici-isys-stream.c    | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ici/ici-isys-frame-buf.c b/drivers/media/pci/intel/ici/ici-isys-frame-buf.c
index ff497f8..39f3ad3 100644
--- a/drivers/media/pci/intel/ici/ici-isys-frame-buf.c
+++ b/drivers/media/pci/intel/ici/ici-isys-frame-buf.c
@@ -830,6 +830,12 @@ int ici_isys_frame_buf_add_next(
 		buf->frame_info.frame_planes[0].mem.userptr);
 
 	css_buf->send_irq_sof = 1;
+	css_buf->send_resp_sof = 1;
+	css_buf->send_irq_eof = 1;
+	css_buf->send_resp_eof = 1;
+	css_buf->send_irq_capture_ack = 1;
+	css_buf->send_irq_capture_done = 1;
+
 	css_buf->output_pins[buf_list->fw_output].addr =
 		(uint32_t)buf->kframe_info.planes[0].dma_addr;
 	css_buf->output_pins[buf_list->fw_output].out_buf_id =
diff --git a/drivers/media/pci/intel/ici/ici-isys-stream.c b/drivers/media/pci/intel/ici/ici-isys-stream.c
index 33fec29..f5b6e39 100644
--- a/drivers/media/pci/intel/ici/ici-isys-stream.c
+++ b/drivers/media/pci/intel/ici/ici-isys-stream.c
@@ -545,7 +545,7 @@ static int start_stream_firmware(struct ici_isys_stream *as)
 
 	reinit_completion(&ip->stream_start_completion);
 	rval = ipu_lib_call(stream_start, as->isys, ip->stream_handle,
-				   NULL);
+				   &css_buf);
 	if (rval < 0) {
 		dev_err(dev, "can't start streaming (%d)\n", rval);
 		goto out_stream_close;
-- 
2.7.4

